akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -237,7 +236,8 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
     def handle(event: StreamFrameEvent): StreamState =
       if (event.isInstanceOf[ParsedHeadersFrame] && streamStates.size > settings.maxConcurrentStreams) {
         // When trying to open a new Stream, if that op would exceed the maxConcurrentStreams, then refuse the op
-        resetStream(event.streamId, ErrorCode.REFUSED_STREAM)
+        debug("Peer trying to open stream that would exceed `maxConcurrentStreams`, refusing stream")
```

Comment:
Good addition. Could this be `info`, though?

Reply:
I considered that but since it is somewhat expected that servers run into that case for some clients, I'd rather avoid spamming the log with client-induced log messages.